src/modules/html.c

```diff
@@ -60,20 +60,20 @@ static unsigned int html_reply(struct http_request *request, void *data)
 	GET_PRIV(data, html);
 	const char *url_stub = (strlen(request->url) > strlen("/html/")) ? request->url + strlen("/html/") : "index.html";
 	if (url_stub[0] == '/' || strstr(url_stub,"/../") || !strncmp(url_stub,"../",strlen("../"))) {
-		send_response_fail(request->connection, "Invalid URL");
+		send_response_bad_request(request->connection, "Invalid URL");
```

Comment:
This is also wrong. What we test for here are valid urls, _we_ just don't accept them.